### PR TITLE
[KBV-399] Refactor to remove address specific functionality

### PR DESCRIPTION
**Pull request by KaraokeStu**
### What changed

- Renamed any references to cri.address -> cri.common
- Removed address specific functionality
- Added spotless checks to pre-merge
- General code tidy (removed unused aliases and classes)
- Updated to use AWS BOM (preserves AWS version dependency better)

### Why did it change

As part of our quick refactoring, the common libraries were still surfaced under the wrong namespace.
This has now been cleaned up.

Also, we found that there were time when you needed a lib specific to the CRI (in this case, the Address Service, Item, and Exception are used/thrown within both the Credential Issuer lambda and the Address save lambda, so these were moved to a CRI-specific lib

CanonicalAddress has been left in the common library as this is also used in the KBV CRI

### Issue tracking
- [KBV-399](https://govukverify.atlassian.net/browse/KBV-399)

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -39,9 +37,11 @@ subprojects {
 		maven {
 			url 'https://gds.jfrog.io/artifactory/di-allowed-repos'
 		}
+		mavenCentral()
```
Comment (jkunle) [inline build.gradle:40]: ? why mavenCentral

Comment (KaraokeStu) [reply build.gradle]: Removed

**Review by idrop — APPROVED** [review]

**Review by idrop — APPROVED** [review]

File: .gitmodules [hunk 2]
```diff
@@ -1,5 +1,5 @@
-[submodule "lib"]
-	path = lib
+[submodule "commmon-lib"]
```
[the remark this answers is not in the document]
Comment (KaraokeStu) [reply .gitmodules]: Done - please re-approve

**Review by idrop — CHANGES_REQUESTED** [review]
putting a hold until we can get an event change in

File: lib/src/main/java/uk/gov/di/ipv/cri/address/library/persistence/item/AddressItem.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    @DynamoDbPartitionKey()
+    public UUID getSessionId() {
+        return sessionId;
+    }
+
+    public void setSessionId(UUID sessionId) {
+        this.sessionId = sessionId;
+    }
+
+    // @DynamoDbConvertedBy(ListOfMapConverter.class)
```
Comment (jkunle) [inline lib/src/main/java/uk/gov/di/ipv/cri/address/library/persistence/item/AddressItem.java:26]: Is this comment mean't to be here or removed

Comment (KaraokeStu) [reply lib/src/main/java/uk/gov/di/ipv/cri/address/library/persistence/item/AddressItem.java]: Removed

**Review by jkunle — DISMISSED** [review]

**Review by idrop — APPROVED** [review]

**Pull request merged**